src/components/Block/SpaceFormProfile.vue

```diff
@@ -16,67 +16,79 @@ const props = withDefaults(
 const emit = defineEmits<{
   (e: 'errors', value: any);
   (e: 'pick', field: any);
+  (e: 'no-network');
 }>();
 
 const availableNetworks = enabledNetworks.map(id => ({
   id,
   name: getNetwork(id).name
 }));
 
-const definition = {
-  type: 'object',
-  title: 'Space',
-  additionalProperties: false,
-  required: ['name', 'walletNetwork', 'walletAddress'],
-  properties: {
-    name: {
-      type: 'string',
-      title: 'Name',
-      minLength: 1,
-      examples: ['Space name']
-    },
-    description: {
-      type: 'string',
-      format: 'long',
-      title: 'About',
-      examples: ['Space description']
-    },
-    externalUrl: {
-      type: 'string',
-      title: 'Website',
-      examples: ['Website URL']
-    },
-    github: {
-      type: 'string',
-      title: 'GitHub',
-      examples: ['GitHub handle']
-    },
-    twitter: {
-      type: 'string',
-      title: 'Twitter',
-      examples: ['Twitter handle']
-    },
-    discord: {
-      type: 'string',
-      title: 'Discord',
-      examples: ['Discord handle']
-    },
-    walletNetwork: {
-      type: 'string',
-      enum: enabledNetworks,
-      options: availableNetworks,
-      title: 'Treasury network'
-    },
-    walletAddress: {
-      type: 'string',
-      format: 'address',
-      title: 'Treasury address',
-      examples: ['0x0000…']
+const definition = computed(() => {
+  return {
+    type: 'object',
+    title: 'Space',
+    additionalProperties: false,
+    required: ['name', 'walletNetwork', 'walletAddress'],
+    properties: {
+      name: {
+        type: 'string',
+        title: 'Name',
+        minLength: 1,
+        examples: ['Space name']
+      },
+      description: {
+        type: 'string',
+        format: 'long',
+        title: 'About',
+        examples: ['Space description']
+      },
+      externalUrl: {
+        type: 'string',
+        title: 'Website',
+        examples: ['Website URL']
+      },
+      github: {
+        type: 'string',
+        title: 'GitHub',
+        examples: ['GitHub handle']
+      },
+      twitter: {
+        type: 'string',
+        title: 'Twitter',
+        examples: ['Twitter handle']
+      },
+      discord: {
+        type: 'string',
+        title: 'Discord',
+        examples: ['Discord handle']
+      },
+      walletNetwork: {
+        type: ['string', 'null'],
+        enum: [...enabledNetworks, null],
+        options: [...availableNetworks, { id: null, name: 'No treasury' }],
+        title: 'Treasury network',
+        nullable: true
+      },
+      walletAddress: {
+        type: props.form.walletNetwork === null ? 'null' : 'string',
+        format: 'address',
+        title: 'Treasury address',
+        examples: ['0x0000…'],
+        hidden: props.form.walletNetwork === null
+      }
     }
+  };
+});
+
+watch(
+  () => props.form.walletNetwork,
+  to => {
+    if (to === null) emit('no-network');
   }
-};
+);
```

Comment:
Why do we need it? Wouldn't normal form updates handle it already?

Reply:
When we set walletNetwork to null we need to set walletAddress to null to, and it can be made outside SpaceFormProfile.vue

Reply:
Makes sense, can we handle it internally though, so it works out of the box?

For example now it won't work in edit space modal.

Reply:
Hm, it is not an option to mutate props.form, so need another way

Reply:
@j0hnfl0w good point, let's keep it as-is then, but fix edit space modal to also handle this event.